opm/autodiff/SimulatorFullyImplicitBlackoilOutput.hpp

```diff
@@ -375,5 +413,214 @@ namespace Opm
     }
 
 
+
+
+
+    namespace detail {
+
+        /**
+         * Converts an ADB into a standard vector by copy
+         */
+        inline std::vector<double> adbToDoubleVector(const Opm::AutoDiffBlock<double>& input) {
+            const auto& b_v = input.value();
+            std::vector<double> b(b_v.data(), b_v.data() + b_v.size());
+            return b;
+        }
+
+
+        template<class Model>
+        std::vector<data::CellData> getCellData(
+                const Opm::PhaseUsage& phaseUsage,
+                const Model& model,
+                const RestartConfig& restartConfig,
+                const int reportStepNum) {
+
+
+            std::vector<data::CellData> simProps;
+
+            //Get the value of each of the keys
+            std::map<std::string, int> outKeywords = restartConfig.getRestartKeywords(reportStepNum);
+            for (auto& keyValue : outKeywords) {
+                keyValue.second = restartConfig.getKeyword(keyValue.first, reportStepNum);
+            }
+
+            const typename Model::SimulatorData& sd = model.getSimulatorData();
+
+            //Get shorthands for water, oil, gas
+            const int aqua_active = phaseUsage.phase_used[Opm::PhaseUsage::Aqua];
+            const int liquid_active = phaseUsage.phase_used[Opm::PhaseUsage::Liquid];
+            const int vapour_active = phaseUsage.phase_used[Opm::PhaseUsage::Vapour];
+
+            const int aqua_idx = phaseUsage.phase_pos[Opm::PhaseUsage::Aqua];
+            const int liquid_idx = phaseUsage.phase_pos[Opm::PhaseUsage::Liquid];
+            const int vapour_idx = phaseUsage.phase_pos[Opm::PhaseUsage::Vapour];
+
+
+            /**
+             * Formation volume factors for water, oil, gas
+             */
+            if (aqua_active && outKeywords["BW"] > 0) {
+                outKeywords["BW"] = 0;
+                simProps.emplace_back(
+                        "1OVERBW",
+                        Opm::UnitSystem::measure::water_formation_volume_factor,
```

Comment:
This will be wrong I think: Opm::UnitSystem::measure::water_formation_volume_factor gives the unit for `B`, the formation volume factor, not `1/B`.

I should have realized this before, sorry for the extra work!